apps/challenges/urls.py

```diff
@@ -40,7 +40,8 @@
     ),
     # `A-Za-z` because it accepts either of `all, future, past or present` in either case
     url(
-        r"^challenge/(?P<challenge_time>[A-Za-z]+)/(?P<challenge_approved>[A-Za-z]+)/(?P<challenge_published>[A-Za-z]+)$",
+        r"^challenge/time/(?P<challenge_time>[A-Za-z]+)/approved/"
+        "(?P<challenge_approved>[A-Za-z]+)/visibility/(?P<challenge_published>[A-Za-z]+)$",
         views.get_all_challenges,
         name="get_all_challenges",
```

Comment:
The view for this API also needs to be changed to only return approved and present challenges by default. If the request params have other values for these flags then we will use that.